teps/0048-task-results-without-results.md

```diff
@@ -283,6 +284,35 @@ spec:
       value: $(tasks.propagate-image-name.results.image)
 ```
 
+#### Providing default values for optional `Results`
+
+Today, a `Task` can declare a `Result` but not produce the `Result`
+and it's execution will still be successful. If a subsequent `Task`
+attempts to use the `Result`, then it would fail (causing the `Pipeline` to fail).
+As such, a `Result` is optional when unused but required when used - this
+can be confusing to users when failures happen later in the `Pipeline`.
+Supporting default values for the optional `Results` would help prevent 
+unexpected failures, and improve our `Task` interface`.
```

Comment:
> would help prevent unexpected failures

Conversely, having a Task-defined default could introduce unexpected successes.

As a user of this example Task, I might want to fall back to the "dev" branch but the author's default is "main". How could I handle this?

Reply:
been thinking of default results as an authoring time concern, similarly to how we do for default parameters (defined in tasks/pipelines) -- but I do see your issue here and maybe we should make it a runtime concern (defined in taskruns/pipelineruns), what do you think?

cc @vinamra28 @pritidesai

Reply:
From my pov I think runtime defaulting makes the most sense, since it's only at that moment the end-user is able to configure what the correct value is for them. However, I can see benefits in both directions and I don't feel super strongly if we decide that author-time defaults are the best approach.